### PR TITLE
Standardize database response and database create request

### PR DESCRIPTION
**Pull request by tenyo**
- Created new `DatabaseCreateRequest` with custom `CreateDBClusterInput` and `CreateDBInstanceInput` structs
- Created a standard `DatabaseResponse` which is now returned by the create, modify and delete operations

### CODE REVIEW

File: actions/orchestration.go [hunk 1]
```diff
@@ -40,70 +45,123 @@ func (o *rdsOrchestrator) databaseCreate(c buffalo.Context, input *DatabaseCreat
...
 			}
 			return nil, ErrCode("failed to create database instance", err)
 		}
 
 		log.Println("Created RDS instance", instanceOutput)
+		instance = instanceOutput.DBInstance
 	}
 
-	return &DatabaseCreateOutput{clusterOutput, instanceOutput}, nil
+	return &DatabaseResponse{cluster, instance}, nil
```
Comment (fishnix) [inline actions/orchestration.go]: This is a small thing, but can we use the more explicit form except in cases where you're defining a struct inline?

File: actions/orchestration.go [hunk 2]
```diff
@@ -179,32 +239,34 @@ func (o *rdsOrchestrator) databaseModify(c buffalo.Context, id string, input *Da
...
+				Tags:         toRDSTags(normalizedTags),
 			}); err != nil {
 				return nil, ErrCode("failed to add tags to database", err)
 			}
 			log.Println("Updated tags for RDS resource", arn)
 		}
 	}
 
-	return &DatabaseModifyOutput{clusterOutput, instanceOutput}, nil
+	return &DatabaseResponse{cluster, instance}, nil
```
Comment (fishnix) [inline actions/orchestration.go]: same

File: actions/orchestration.go [hunk 3]
```diff
@@ -301,26 +365,8 @@ func (o *rdsOrchestrator) databaseDelete(c buffalo.Context, id string, snapshot
...
-	}
-
-	normalizedTags = append(normalizedTags,
-		&rds.Tag{
-			Key:   aws.String("spinup:org"),
-			Value: aws.String(Org),
-		})
-
-	return normalizedTags
+	return &DatabaseResponse{cluster, instance}, nil
```
Comment (fishnix) [inline actions/orchestration.go]: same

File: actions/types.go [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+type DatabaseCreateRequest struct {
+	Cluster  *CreateDBClusterInput
+	Instance *CreateDBInstanceInput
+}
+
+func (dcr DatabaseCreateRequest) String() string {
+	s, _ := json.MarshalIndent(dcr, "", "\t")
+	return string(s)
+}
```
Comment (fishnix) [inline actions/types.go:18]: You define this here but in the orchestrator you use `%+v` - you probably don't need this.

File: actions/types.go [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+	EngineVersion               *string
+	MasterUserPassword          *string
+	MasterUsername              *string
+	MultiAZ                     *bool
+	Port                        *int64
+	PubliclyAccessible          *bool
+	StorageEncrypted            *bool
+	Tags                        []*Tag
+	VpcSecurityGroupIds         []*string
+}
```
Comment (fishnix) [inline actions/types.go:40]: Do all of these things actually come from the user? offering?  I see a few I wonder about -- 
* AutoMinorVersionUpgrade
* EnableCloudwatchLogsExports
* PubliclyAccessible
* StorageEncrypted

File: actions/types.go [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+	EngineVersion               *string
+	MasterUserPassword          *string
+	MasterUsername              *string
+	Port                        *int64
+	ScalingConfiguration        *ScalingConfiguration
+	SnapshotIdentifier          *string
+	StorageEncrypted            *bool
+	Tags                        []*Tag
+	VpcSecurityGroupIds         []*string
+}
```
Comment (fishnix) [inline actions/types.go:60]: same here

**Review by fishnix — APPROVED** [review]
Just those comments, otherwise looks good! 👍

File: actions/types.go (hunk 5 above)
Comment (tenyo) [reply actions/types.go:40]: These come from the UI DB Controller

File: actions/orchestration.go (hunk 1 above)
Comment (tenyo) [reply actions/orchestration.go]: What more explicit form?

File: actions/types.go (hunk 4 above)
Comment (tenyo) [reply actions/types.go:18]: It works. Without it I get `&{Cluster:0xc0003560c0 Instance:<nil>}` when I print the struct

File: actions/types.go (hunk 5 above)
Comment (tenyo) [reply actions/types.go:40]: Replaced some of those with opinionated defaults

File: actions/types.go (hunk 4 above)
Comment (fishnix) [reply actions/types.go:18]: oh that's interesting...  I'll have to try that.  i always use `awsutil.PrettyPrint()` when I want those details, maybe that's all it does. 🤯

File: actions/types.go (hunk 5 above)
Comment (fishnix) [reply actions/types.go:40]: 🎉

**Pull request merged**